package.json

```diff
@@ -19,10 +19,11 @@
     "ci": "npm run lint && npm test",
     "lint": "eslint ./src --ext .ts",
     "prepublishOnly": "npm run build",
-    "build": "rm -rf dist && npm run build:cjs && npm run build:esm && attw --pack .",
+    "build": "rm -rf dist && npm run build:check-types && npm run build:cjs && npm run build:esm && attw --pack .",
+    "build:check-types": "tsc -p tsconfig.json --noEmit",
```

Comment:
`types/` wasn't previously being typechecked; I introduced this script and reconfigured the `tsconfig.*.json` files such that they would be.